libimage/runtime.go

```diff
@@ -574,23 +574,30 @@ func (r *Runtime) ListImages(ctx context.Context, names []string, options *ListI
 		options = &ListImagesOptions{}
 	}
 
-	var images []*Image
-	if len(names) > 0 {
-		for _, name := range names {
-			image, _, err := r.LookupImage(name, nil)
-			if err != nil {
-				return nil, err
-			}
-			images = append(images, image)
-		}
-	} else {
-		storageImages, err := r.store.Images()
+	lookupImagesMap := make(map[string]struct{}, 0)
+	for _, name := range names {
+		image, _, err := r.LookupImage(name, nil)
 		if err != nil {
 			return nil, err
 		}
-		for i := range storageImages {
-			images = append(images, r.storageToImage(&storageImages[i], nil))
+		lookupImagesMap[image.ID()] = struct{}{}
+	}
+
+	storageListResult, err := r.store.MultiList(
```

Comment:
(This works just fine; calling the result maybe just `snapshot` might be a bit shorter.)